### PR TITLE
Add_column proxies to mutate

### LINKED ISSUE #1291: add_column has different behavior than mutate
```python
t = ibis.table([('a', 'int64')], name='t')
t.add_column(ibis.timestamp('now').cast('date'), name='date')  # fails
t.mutate(date=ibis.timestamp('now').cast('date'))  # works
```

@kszucs do you think you could knock this one out? `add_column`  should probably just call `mutate`

### PR DESCRIPTION
**Pull request by kszucs**
closes #1291

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -2683,6 +2683,21 @@ def _table_materialize(table):
         return TableExpr(op)
 
 
+def add_column(table, expr, name=None):
```
Comment (cpcloud) [inline ibis/expr/api.py]: I think we should deprecate this functionality after 1 or 2 more release cycles. It's pretty redundant.

File: ibis/expr/tests/test_table.py [hunk 2]
```diff
@@ -1240,6 +1240,12 @@ def g(x):
     assert_equal(result, expected)
 
 
+def test_add_column_proxies_to_mutate(table):
+    result = table.add_column(ibis.now().cast('date'), name='date')  # fails
```
Comment (cpcloud) [inline ibis/expr/tests/test_table.py]: does this actually fail, or what does that comment mean?

Comment (kszucs) [reply ibis/expr/tests/test_table.py]: Just a copy paste from the issue example, fixing it.

File: ibis/expr/api.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/api.py]: That seems like a useful tool. I'm okay with introducing that as a dependency.

**Comment by cpcloud** [comment]
@kszucs can you rebase? will merge on green

**Comment by kszucs** [comment]
@cpcloud rebase done

**Comment by cpcloud** [comment]
thanks @kszucs

**Pull request closed without merging**